### ISSUE TITLE
Correctly replace knitr options which had their name normalized from - to .

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#5852]
Example from 

* https://github.com/quarto-dev/quarto-cli/discussions/5813
* Other occurence: 
  * https://github.com/quarto-dev/quarto-cli/issues/6491

This below won't work 
````markdown
---
format: html
---

```{r}
#| tidy: true
#| tidy-opts: !expr list(width.cutoff = 50)
# a long expression
1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+
1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1
```
````

This is because 

* knitr defines `tidy.opts = NULL` as a default
* knitr will parse the YAML chunk option and sees `tidy-opts` so it won't replace the default. Options have both now `tidy-opts` and `tidy.opts`
* Quarto will normalize `-` to `.` in an option hook step, but this will just rename. This means options in knitr have twice `tidy.opts` with two different value. 
* Knitr will use only the first value, which will be the default one. 

Our options hooks should make the replacement. 

Longer term, **knitr** should probably do the normalization as the YAML chunk option parsing step

https://github.com/quarto-dev/quarto-cli/blob/5df9451723c7fe81462700c19709c0b06846de95/src/resources/rmd/hooks.R#L11-L18

https://github.com/quarto-dev/quarto-cli/blob/5df9451723c7fe81462700c19709c0b06846de95/src/resources/rmd/hooks.R#L631-L634

**Comment by yihui** [comment]
I think we can normalize all option names inside **knitr**. I'll take a closer look today.

**Comment by cderv** [comment]
Ok great ! Here is the branch I had in case we wanted to fix both issues in Quarto 
https://github.com/quarto-dev/quarto-cli/compare/main...fix%2Fknitr%2Foption-normalization

It seems better to me to do it in **knitr**, even if this means that user's would have to update **knitr**.

**Comment by cderv** [comment]
This has been fixed on **knitr** - from next version 1.44 (and from current dev version 1.43.7), all dash option are changed to their dot version. This includes `tidy-opts` and `cache-lazy`
